### PR TITLE
MOB-2047 - Setup mpc wallet after purchase during onboarding

### PR DESCRIPTION
**Pull request by Oleg-Pecheneg**

### CODE REVIEW

File: unstoppable-ios-app/domains-manager-ios/Modules/MPC/PurchaseMPCFlow/Checkout/MPCOnboardingPurchaseCheckoutViewController.swift [hunk 1]
```diff
@@ -13,8 +13,9 @@ final class MPCOnboardingPurchaseCheckoutViewController: BaseViewController, Vie
     override var preferredStatusBarStyle: UIStatusBarStyle { .default }
     
     weak var onboardingFlowManager: OnboardingFlowManager?
-    var progress: Double? { 3 / 4 }
-    
+    var dashesProgressConfiguration: DashesProgressView.Configuration { .init(numberOfDashes: 3) }
+    var progress: Double? { 0.5 }
```
Comment (rommex) [inline unstoppable-ios-app/domains-manager-ios/Modules/MPC/PurchaseMPCFlow/Checkout/MPCOnboardingPurchaseCheckoutViewController.swift:17]: 1/2 ? 2/4? for consistency )

File: unstoppable-ios-app/domains-manager-ios/Modules/MPC/PurchaseMPCFlow/MPCWalletPasswordValidator.swift [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+extension MPCWalletPasswordValidator {
+    var minMPCWalletPasswordLength: Int { 12 }
+    var maxMPCWalletPasswordLength: Int { 32 }
+    
+    func validateWalletPassword(_ password: String) -> [MPCWalletPasswordValidationError] {
+        let minLength = minMPCWalletPasswordLength
+        let maxLength = maxMPCWalletPasswordLength
+        let numberRegex = try! NSRegularExpression(pattern: "\\d")
+        let specialCharacterRegex = try! NSRegularExpression(pattern: "[!@#$%^&*()_+\\-\\[\\]{};':\"\\\\|,.<>\\/?]+")
```
Comment (rommex) [inline unstoppable-ios-app/domains-manager-ios/Modules/MPC/PurchaseMPCFlow/MPCWalletPasswordValidator.swift:22]: Please extract to `Constants.swift` and locate near `ETHRegexPattern`

File: unstoppable-ios-app/domains-manager-ios/SupportingFiles/Localization/en.lproj/Localizable.strings [hunk 3]
```diff
@@ -1136,6 +1137,19 @@ More tabs are coming in the next updates.";
 "MPC_WALLET_ALREADY_PURCHASED_TITLE" = "Looks like you already have Unstoppable Wallet";
 "MPC_WALLET_ALREADY_PURCHASED_SUBTITLE" = "We've extended your subscription by 1 year based on your payment. Import your Unstoppable Wallet to start using it with the app, or use a different email to create a new Unstoppable Wallet.";
 "USE_DIFFERENT_EMAIL" = "Use different email";
+"MPC_PASSWORD_VALIDATION_LENGTH_TITLE" = "Minimum %i characters";
+"MPC_PASSWORD_VALIDATION_TOO_LONG_TITLE" = "Minimum %i characters, maximum %i characters";
+"MPC_PASSWORD_VALIDATION_NUMBER_TITLE" = "At least one number";
+"MPC_PASSWORD_VALIDATION_SPECIAL_CHAR_TITLE" = "At least one special character (e.g. -!&*)";
+"MPC_TAKEOVER_CREDENTIALS_SUBTITLE" = "Each Unstoppable Wallet requires a UNIQUE email address. Make sure you have access to an email. It can be the one used in the previous step or a different one.";
```
Comment (rommex) [inline unstoppable-ios-app/domains-manager-ios/SupportingFiles/Localization/en.lproj/Localizable.strings:1144]: a -> the: `Make sure you have access to the email.`

**Review by rommex — APPROVED** [review]
With minor changes

File: unstoppable-ios-app/domains-manager-ios/Modules/MPC/PurchaseMPCFlow/MPCWalletPasswordValidator.swift (hunk 2 above)
Comment (Oleg-Pecheneg) [reply unstoppable-ios-app/domains-manager-ios/Modules/MPC/PurchaseMPCFlow/MPCWalletPasswordValidator.swift:22]: This regex rules is specifically to satisfy BE requirements when create password for MPC wallet. 
I'm not sure it needs to be on the global Constants level.

**Pull request merged**